### PR TITLE
Custom vision 3 1

### PR DESCRIPTION
**Pull request by areddish**
### Latest improvements:
Adding v3.1 of Custom Vision Training API
  - Introduces a Suggestion API surface that allows for users to get suggested tags and regions in their untagged images. Also supporting query functions for finding image counts and images that were suggested.
  - Added TensorFlowLite export support
 
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/cognitiveservices/data-plane/CustomVision/Training/stable/v3.1/Training.json [hunk 1]
```diff
@@ -0,0 +1,5046 @@
...
+                "consumes": [],
+                "produces": [
+                    "application/json",
+                    "application/xml",
+                    "text/xml"
+                ],
+                "parameters": [
+                    {
+                        "name": "projectId",
+
```
Comment (nschonni) [inline specification/cognitiveservices/data-plane/CustomVision/Training/stable/v3.1/Training.json]:
```suggestion
        "/projects/{projectId}/tagsAndRegions/suggestions": {
```

File: specification/cognitiveservices/data-plane/CustomVision/Training/stable/v3.1/Training.json [hunk 2]
```diff
@@ -0,0 +1,5046 @@
...
+        "https"
+    ],
+    "paths": {
+        "/domains": {
+            "get": {
+                "tags": [
+                    "DomainsApi"
+                ],
+                "summary": "Get a list of the available domains.",
+                "operationId": "GetDomains",
```
Comment (yungezz) [inline specification/cognitiveservices/data-plane/CustomVision/Training/stable/v3.1/Training.json]: normally operationId is like `Resource_Action`

Comment (areddish) [reply specification/cognitiveservices/data-plane/CustomVision/Training/stable/v3.1/Training.json]: We've not been using that style of naming in cognitive serivces. Renaming the operations isn't something we'd want to do in this change as we're trying not to break existing clients.

**Comment by yungezz** [comment]
https://github.com/Azure/azure-rest-api-specs-pr/pull/817

**Comment by yungezz** [comment]
hi @ruowan could you pls help to look at the avocado error? thanks.

**Comment by ruowan** [comment]
Hi, @areddish 
The avocado error report is `Traning.json` is not a valid json(JS standard library can't parse it), because `Traning.json` format is `utf-8 with Bom`  which causes adding extra BOM marker such as `0xfeff`  at the beginning of the file. To solve this issue, you need to use notepad or other editor open `Traning.json` and save as `uft-8` format instead of `utf-8 with BOM`.

**Comment by yungezz** [comment]
The PR is ready to merge after CI fixed. PR is reviewed by ARM team in private repo already.

**Comment by yungezz** [comment]
@areddish  pls fix CI per @ruowan 's comments. then the  PR is ready to go.

**Comment by areddish** [comment]
@yungezz - I've updated the encoding and it appears all is well with the checks.

**Comment by yungezz** [comment]
hi @areddish , there're still 2 errors in  CI validation. could you pls do below fixing:
- for Spellcheck,  pls add `tagsandregions` to root https://github.com/Azure/azure-rest-api-specs/blob/master/custom-words.txt file.
- for ModelValidation, in line https://github.com/Azure/azure-rest-api-specs/pull/7074/files#diff-f36f6dba529475ad82fb8bb8ba7869abR1027,  response 200 definition, could you pls follow sample in previous version: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/cognitiveservices/data-plane/CustomVision/Training/stable/v3.0/Training.json#L211-L215. 

I'll merge it after those issues fixed and CI passed.

**Review by yungezz — APPROVED** [review]

**Pull request merged**